backend/subroker/views.py

```diff
@@ -312,28 +312,118 @@ def group_add_user(request, group_id):
+            params={'api_key':'e50a928b29c2b7ed1f82ba2e6ce4982c','query':search_str.replace(" ","+")}
+            r = requests.get(url, params = params)
+            if r.status_code == 200:
+                data = r.json()
+                return JsonResponse(data, status=200)
+            else:
+                attempt_num += 1
+                # Wait for 5 seconds before re-trying
+                time.sleep(5)  
+        return HttpResponse(status=405)
+    else:
+        return HttpResponseNotAllowed(['GET'])
+
 def content_detail(request, content_id):
     if request.method == 'GET':
         if request.user.is_authenticated:
-            try:
-                content = Content.objects.get(id=content_id)
-            #ERR 404 : Content Doesn't Exist
-            except (Content.DoesNotExist) as e:
-                return HttpResponse(status=404)
-            fav_users = [user for user in content.favorite_users.all().values()]
-            return JsonResponse({
-                "id": content.id, 
-                "the_movie_id": content.the_movie_id,
-                "name": content.name,
-                "favorite_count": content.favorite_cnt,
-                "favorite_users": fav_users
-                }, status=200)
+            # keep track of how many times we've retried
+            MAX_RETRIES = 2
+            attempt_num = 0  
+
+            #get content from TMDB
+            while attempt_num < MAX_RETRIES:
+                url = 'https://api.themoviedb.org/3/movie/' + str(content_id)
+                params={'api_key':'e50a928b29c2b7ed1f82ba2e6ce4982c'}
+                r = requests.get(url, params = params)
+                #got content from TMDB
+                if r.status_code == 200:
+                    data = r.json()
+                    id = data['id']
+                    name = data['title']
+                    genres = []
+                    for genre in data['genres'][: -1]:
+                        genres.append(genre['name'] + ", ")
+                    genres.append(data['genres'][-1]['name'])
+                    countries = []
+                    for country in data['production_countries'][: -1]:
+                        countries.append(country['name'] + ", ")
+                    countries.append(data['production_countries'][-1]['name'])
+                    poster = data['poster_path']
+                    overview = data['overview']
+                    release_date = data['release_date']
+                    rate = data['vote_average']
+                    #check if the content is in DB
+                    try:
+                        content = Content.objects.get(id=content_id)
+                    #Content is not in our DB
+                    except(Content.DoesNotExist) as e:
+                        create_url = 'http://localhost:8000/api/content/' +  str(content_id) + '/'
+                        r = requests.post(create_url, headers=request.headers)
```

Comment:
@muhwagwa How about creating a new post here?
```suggestion
                        content = Content()
                        content.id = id
                        content.save()
```